solutions/swb-app/src/projectRoutes.ts

```diff
@@ -3,20 +3,44 @@
  *  SPDX-License-Identifier: Apache-2.0
  */
 
-import { ProjectService } from '@aws/workbench-core-accounts';
-import CreateProjectRequest from '@aws/workbench-core-accounts/lib/models/createProjectRequest';
+import { ProjectService, CreateProjectRequest, ListProjectsRequest } from '@aws/workbench-core-accounts';
+import Boom from '@hapi/boom';
 import { Request, Response, Router } from 'express';
 import { validate } from 'jsonschema';
 import { wrapAsync } from './errorHandlers';
 import CreateProjectSchema from './schemas/projects/createProjectSchema';
+import ListProjectsSchema from './schemas/projects/listProjectsSchema';
 import { processValidatorResult } from './validatorHelper';
 
 export function setUpProjectRoutes(router: Router, projectService: ProjectService): void {
-  // Get projects
+  // List projects
   router.get(
     '/projects',
     wrapAsync(async (req: Request, res: Response) => {
-      const projects = await projectService.listProjects();
+      let projects = {};
+      // validate if query params present
+      if (req.query.pageSize || req.query.paginationToken) {
+        const objectToValidate = {
+          pageSize: req.query.pageSize ? Number(req.query.pageSize) : undefined,
+          paginationToken: req.query.paginationToken
+        };
+        processValidatorResult(validate(objectToValidate, ListProjectsSchema));
+        // Apply pagination if applicable
+        if (
+          (objectToValidate.paginationToken && typeof objectToValidate.paginationToken !== 'string') ||
```

Comment:
Is it possible to confirm that schema validation will take care of types? E.g. could we try to send number and see if it's rejected by schema validation and not the code